File: src/commonMain/kotlin/com/scoreo/domain/port/PlayerRepository.kt
package com.scoreo.domain.port

import com.scoreo.domain.model.Player

interface PlayerRepository {
    fun getAll(includeInactive: Boolean = false): List<Player>
    fun save(player: Player)
    fun saveAll(players: List<Player>)
    fun delete(id: String, anonymize: Boolean = false)
}

File: apps/scoreo/src/domain/port/playerRepository.ts
import type { Player } from '../model/player'

export interface PlayerRepository {
  getAll(includeInactive?: boolean): Player[]
  save(player: Player): void
  saveAll(players: Player[]): void
  delete(id: string, anonymize?: boolean): void
  hardDelete(id: string): void
  deleteAll(): void
}
